viewer/src/models/cad/filterCurrentWantedSectors.ts

```diff
@@ -0,0 +1,24 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import { Observable, of, pipe, OperatorFunction, empty } from 'rxjs';
+import { flatMap, withLatestFrom } from 'rxjs/operators';
+import { ParsedSector } from '../../data/model/ParsedSector';
+import { WantedSector } from '../../data/model/WantedSector';
+
+export function filterCurrentWantedSectors(
+  wantedObservable: Observable<WantedSector[]>
+): OperatorFunction<ParsedSector, ParsedSector> {
+  return pipe(
+    withLatestFrom(wantedObservable),
+    flatMap(([loaded, wanted]) => {
+      for (const wantedSector of wanted) {
+        if (loaded.id === wantedSector.id && loaded.levelOfDetail === wantedSector.levelOfDetail) {
+          return of(loaded);
+        }
+      }
+      return empty();
+    })
```

Comment:
Is this a O(N^2) matching search?

Reply:
Depends on how you look at it, I guess. For each sector that has been loaded (fetched and parsed), we go through the list of wanted and see if it is still there at the correct level of quality. So it is `O(N)` when it is run, but there are of course also `N` sectors incoming, but that is stretched over time, if it helps.